src/panoptes/pocs/state/machine.py

```diff
@@ -143,10 +143,14 @@ def run(self, exit_when_done=False, run_once=False, initial_next_state='ready'):
                 state_changed = self.goto_next_state()
             except Exception as e:
                 self.logger.critical(f"Problem going from self.state={self.state!r} to "
-                                     f" self.next_state={self.next_state!r}"
-                                     f", exiting loop [{e!r}]")
-                # TODO should we automatically park here?
-                self.stop_states()
+                                     f" self.next_state={self.next_state!r}: {e}. Parking.")
+                try:
+                    self.next_state = "parking"
+                    self.goto_next_state()
+                except Exception as e:
+                    self.logger.critical(f"Problem while parking after state machine error: {e}.")
+                finally:
+                    self.stop_states()
```

Comment:
So this would stop the state machine after the transition to "parking" has completed (i.e. the state's on_enter() has run), preventing it continuing to "parked", "housekeeping", etc.?  Given that an error has occurred this basically does what we want (tells the mount and dome to park, and won't do anything else until restarted) but it would be nice if could enter "parked" before stopping.

Reply:
Good point.